addons/mail/static/src/chatter/web/chatter_patch.js

```diff
@@ -188,24 +170,26 @@ patch(Chatter.prototype, {
     },
 
     async updateRecipients(record, mode = this.state.composerType) {
-        if (!record) {
+        if (!record || !this.state.thread) {
             return;
         }
         const partnerIds = []; // Ensure that we don't have duplicates
         let email;
-        this.mailImpactingFields.recordFields.forEach((field) => {
-            const value = record._changes[field];
-            if (record.data[field] !== undefined && value) {
-                partnerIds.push(value[0]);
-            }
-        });
-        this.mailImpactingFields.emailFields.forEach((field) => {
-            const value = record._changes[field];
-            if (record.data[field] !== undefined && value) {
+        if (this.state.thread.partner_fields) {
+            for (const field of this.state.thread.partner_fields) {
+                const value = record._changes[field];
+                if (record.data[field] !== undefined && value) {
+                    partnerIds.push(value.id);
+                }
+            };
+        }
+        if (this.state.thread.primary_email_field) {
+            const value = record._changes[this.state.thread.primary_email_field];
+            if (record.data[this.state.thread.primary_email_field] !== undefined && value) {
                 email = value;
                 return;
```

Comment:
I don't think this is correct, the return was inside the scope of the `forEach`.

Reply:
The PR that introduced those fields did not test anything besides calling the new route (that I removed). 
I guess I have to build them now